simple_cnn.py

```diff
@@ -0,0 +1,170 @@
+    model.add(layers.Dense(64, activation='relu'))
+    model.add(layers.Dense(num_classes, activation='softmax'))
+
+    model.summary()
+
+    model.compile(optimizer='adam',
+                loss='sparse_categorical_crossentropy',
+                metrics=['accuracy'])
+
+    model.fit(train_images, train_labels, epochs=num_epochs)
+
+    test_loss, test_acc = model.evaluate(test_images, test_labels)
+
+    return test_acc
+
+
+def main(argv):
+    """Main function to run the models
+
+    Args:
+        argv (list): CLI arguments
+            - argv[1] (str): expected to be name of dataset to be used
+            - argv[2] (str): expected to specify the number of runs
+    """
+    number_of_runs = 1
+    if len(argv) > 2:
+        number_of_runs = int(argv[2])
+
+    dataset_name_  = argv[1]
+    if dataset_name_.upper() == "MNIST":
+        input_shape_ = (28, 28, 1)
+        num_classes_ = 10
+        num_epochs_  = 5
+
+        train_images_, train_labels_ = data_loader("mnist", "train", dtype=np.float32)
+        test_images_, test_labels_ = data_loader("mnist", "test", dtype=np.float32)
+        train_labels_ = np.argmax(train_labels_, axis=1) # one-hot-decoding
+        test_labels_  = np.argmax(test_labels_, axis=1)  # one-hot-decoding
+
+    elif dataset_name_.upper() == "CIFAR10":
+        grayscale_   = False
+        input_shape_ = (32, 32, 3)
+        num_classes_ = 10
+        num_epochs_  = 10
+
+        train_images_ = np.empty((0, 32, 32, 3))
+        train_labels_ = np.empty((0, 10))
+
+        for i in range(1, 6):
+            train_images_batch_, train_labels_batch_ = data_loader(
+                "cifar10",
+                "data_batch_" + str(i),
+                dtype=np.float32,
+                use_grayscale=grayscale_)
+
+            train_images_ = np.concatenate((train_images_, train_images_batch_))
+            train_labels_ = np.concatenate((train_labels_, train_labels_batch_))
+
+            test_images_, test_labels_ = data_loader( "cifar10", "test_batch",
+                    dtype=np.float32, use_grayscale=grayscale_)
+
+        train_labels_ = np.argmax(train_labels_, axis=1) # one-hot-decoding
+        test_labels_  = np.argmax(test_labels_, axis=1)  # one-hot-decoding
+
+    elif dataset_name_.upper() == "CIFAR100":
+        grayscale_   = False
+        input_shape_ = (32, 32, 3)
+        num_classes_ = 100
+        num_epochs_  = 12
+
+        train_images_, train_labels_ = data_loader("cifar100", "train",
+                dtype=np.float32, use_grayscale=grayscale_)
+
+        test_images_, test_labels_ = data_loader("cifar100", "test",
+                dtype=np.float32, use_grayscale=grayscale_)
+
+        train_labels_ = np.argmax(train_labels_, axis=1) # one-hot-decoding
+        test_labels_  = np.argmax(test_labels_, axis=1)  # one-hot-decoding
+
+    else:
+        raise Exception("dataset has to be one of 'mnist', 'cifar10' or 'cifar100'.")
+
+    if number_of_runs > 1:
```

Comment:
if you want to be DRY, this can be collapsed, a loop over 1:1 is just the same as your else

Reply:
Yeah, you are right. I thought that ```numpy.mean``` and ```numpy.std``` don't work on 1D arrays with a single entry, but they do. For the Python3 ```statistics``` library this is not the case though.

Reply:
Plesae see 7ca681a9709d9232e8071853dffbe4346a6703d8